pkg/pillar/devicenetwork/devicenetwork.go

```diff
@@ -25,6 +25,17 @@ func IsProxyConfigEmpty(proxyConfig types.ProxyConfig) bool {
 	return false
 }
 
+// IsExplicitProxyConfigured returns true if EVE is explicitly configured
+// to route traffic via a proxy for a given uplink interface.
+func IsExplicitProxyConfigured(proxyConfig types.ProxyConfig) bool {
```

Comment:
Don't we already have some function which checks this?
Or do we have a function which also check for a proxy cert being configured?

Reply:
There is `IsProxyConfigEmpty` above, but that one also checks for transparent proxies, which is not desired in this case.
With transparent proxy, domain name resolution is done by the client, while with explicit proxy it is done by the proxy.
So I needed to differentiate between these two cases for diag's sake.